app/controllers/tags_controller.rb

```diff
@@ -24,32 +24,49 @@ def load_tag_and_subtags
     end
   end
 
+  def reindex
+    work_ids = []
+    unless logged_in_as_admin?
+      flash[:error] = ts('Please log in as admin')
+      redirect_to(request.env['HTTP_REFERER'] || root_path) && return
+    end
+    @tag = Tag.find_by_name(params[:id])
+    work_ids = @tag.work_ids
+    @tag.synonyms.each do |syn|
+      work_ids.push syn.work_ids
+    end
+    work_ids.flatten!
+    reindex_all_works(work_ids)
+    flash[:notice] = ts('Tag sent to be reindexed')
+    redirect_to(request.env['HTTP_REFERER'] || root_path) && return
+  end
+
```

Comment:
I can write a test that presses the button and sees if the right message is displayed but I don’t know how to break things so the fixing it does something useful.